### PR TITLE
Renamed display_ads to billboards: variables and descriptions in tests

### PR DESCRIPTION
**Pull request by lightalloy**
## What type of PR is this? (check all applicable)
- [x] Refactor

## Description
- renamed variables and descriptions in tests
- renamed a `display_ad` factory to `billboard` but kept an alias for `taggable` shared_examples because it relies on a class name which is still `DisplayAd`. We should be able to remove the alias later when we rename the model.
 
## Related Tickets & Documents
- Related Issue #19372 

## QA Instructions, Screenshots, Recordings
Billboards functionality should work as it used to

### CODE REVIEW

**Review by dukegreene — APPROVED** [review]
Looks solid with billboards appearing as normal in the feed and admin UIs. Thanks for modeling how to do big changes like this one wave at a time, and for communicating the possibility of merge conflicts as you ran into them!

File: db/schema.rb [hunk 1]
```diff
@@ -1283,7 +1283,6 @@
     t.index ["reset_password_token"], name: "index_users_on_reset_password_token", unique: true
     t.index ["twitter_username"], name: "index_users_on_twitter_username", unique: true
     t.index ["username"], name: "index_users_on_username", unique: true
-    t.check_constraint "username IS NOT NULL", name: "users_username_not_null"
```
Comment (jaw6) [inline db/schema.rb:1286]: Minor: Possibly this is just GitHub being weird, but this seems like an unrelated change?

**Review by jaw6 — APPROVED** [review]
One odd & minor note in the `schema` aside, this looks great!

File: db/schema.rb (hunk 1 above)
Comment (dukegreene) [reply db/schema.rb:1286]: yeah i just checked main and this change isn’t over there. @jaw6 maybe we could merge this and then do an immediate fast follow to replace the removed line? 

I feel like the validations on the user model itself (ll. 154-154) would be a sturdy enough guard in the meantime.

File: db/schema.rb [hunk 2]
```diff
@@ -1283,7 +1283,6 @@
     t.index ["reset_password_token"], name: "index_users_on_reset_password_token", unique: true
     t.index ["twitter_username"], name: "index_users_on_twitter_username", unique: true
     t.index ["username"], name: "index_users_on_username", unique: true
```
Comment (dukegreene) [inline db/schema.rb:1285]:
```suggestion
    t.index ["username"], name: "index_users_on_username", unique: true
    t.check_constraint "username IS NOT NULL", name: "users_username_not_null"
```

Comment (dukegreene) [reply db/schema.rb:1285]: Seeing if we can merge the suggested change in-place without needing a fast follow

**Review by dukegreene — COMMENTED** [review]
🤞
